FeedAPIChallenge/Feed API/RemoteFeedLoader.swift

```diff
@@ -19,6 +43,53 @@ public final class RemoteFeedLoader: FeedLoader {
 	}
 
 	public func load(completion: @escaping (FeedLoader.Result) -> Void) {
-		fatalError("Must be implemented")
+		client.get(from: url) { [weak self] httpClientResult in
+
+			guard self != nil else { return }
+
+			var result: FeedLoader.Result
+
+			let feedLoaderResult = Self.httpClientResult2RemoteFeedLoaderResult(httpClientResult)
+			switch feedLoaderResult {
+			case .success(let responseData):
+				result = Self.responseData2FeedLoaderResult(responseData)
+
+			case .failure(let error):
+				result = .failure(error)
+			}
+
+			completion(result)
+		}
+	}
+
+	private static func httpClientResult2RemoteFeedLoaderResult(_ httpClientResult: HTTPClient.Result) -> RemoteFeedLoaderResult {
+		let feedLoaderResult = httpClientResult.mapError { _ in Error.connectivity }
+			.flatMap { (responseData, httpResponse) -> RemoteFeedLoaderResult in
+				guard httpResponse.isStatusOK else {
+					return .failure(Error.invalidData)
+				}
+				return .success(responseData)
+			}
+		return feedLoaderResult
+	}
```

Comment:
By changing the `RemoteFeedLoaderResult` to `DataResult`, we have to change it in this function. We can also change its name again to better represent the type returned. Also, there's no need to  create a variable for `feedLoaderResult`, we can return the result of `httpClientResult.mapError.flatMap` directly:
```suggestion
	private static func dataResult(from httpClientResult: HTTPClient.Result) -> DataResult {
		return httpClientResult.mapError { _ in Error.connectivity }
			.flatMap { (responseData, httpResponse) in
				guard httpResponse.isStatusOK else {
					return .failure(Error.invalidData)
				}
				return .success(responseData)
			}
	}
```